src/main/scala/com/ibm/developer/eventstore/akka/OnlineRetailWebServer.scala

```diff
@@ -87,45 +98,26 @@ class OnlineRetailWebServer extends HttpApp {
     }
   }
 
-  override protected def routes: Route =
-    pathSingleSlash {
-      complete {
-        println("GET /")
-        HttpEntity(
-          ContentTypes.`text/html(UTF-8)`,
-          "<html><body>OnlineRetailWebServer is running!</body></html>")
-      }
-    } ~
-      pathPrefix("websocket") {
-        path("orderitem") {
-          println("Incoming ws: websocket/orderitem")
-          handleWebSocketMessages(websocket)
-        }
-      }
-
-  override protected def postHttpBindingFailure(cause: Throwable): Unit =
-    println(s"postHttpBindingFailure: $cause")
-
-  def start(host: String = "localhost", port: Int = 8080): Future[Done] = {
-    val settings = ServerSettings(system.settings.config)
-    Future {
-      startServer(host, port, settings, system)
-    }.map(_ => Done)
-  }
-
-  override protected def waitForShutdownSignal(system: ActorSystem)(implicit ec: ExecutionContext): Future[Done] =
-    shutdownPromise.future
-
-  def stop(): Future[Terminated] = {
-    shutdownPromise.tryComplete(Success(Done))
-    system.terminate()
-  }
+  def routes: Route =
+    extractLog { logger =>
+      concat(
+        pathSingleSlash {
+          logger.info("GET /")
+          complete(HttpEntity(
+            ContentTypes.`text/html(UTF-8)`,
+            "<html><body>OnlineRetailWebServer is running!</body></html>"))
+        },
+        path("websocket" / "orderitem") {
+          logger.info("Incoming ws: websocket/orderitem")
+          extractMaterializer { materializer =>
+            handleWebSocketMessages(websocket(materializer))
+          }
+        })
+    }
 
   def main(args: Array[String]): Unit = {
+    ConfigurationReader.setConnectionEndpoints("0.0.0.0:1100")
     println("Starting OnlineRetailWebServer...")
-    start("localhost", 8080)
+    startServer("localhost", 8080)
```

Comment:
better keep all start logic in one place rather than put one part in the constructor

Reply:
In general better to use logger over println, but here the actor system is not started yet so we can't use the logging infra before that.